docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,378 @@
+    // IPAMClaimReference container the IPAMClaim name where the IPs for this
+    // attachment will be located.
+    IPAMClaimReference string `json:"ipam-claim-reference,omitempty"`
+}
+```
+
+Any other field set in the struct will be ignored by OVN-Kubernetes.
+
+When using the `1.multus-cni.io/default-network` annotation, Multus strictly requires its value to reference an
+existing NAD. Multus then builds the CNI requests based on it.
+This proposal introduces a static default NAD object applied to the cluster. This object will serve as a
+stub to generate the CNI calls, preserving the current behavior:
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: default
+  namespace: ovn-kubernetes
+spec:
+  config: '{"cniVersion": "0.4.0", "name": "ovn-kubernetes", "type": "ovn-k8s-cni-overlay"}'
+```
+With this approach, users must configure the `Name` to `default` and the `Namespace` to `ovn-kubernetes`.
+This configuration ensures Multus still references the default network while OVN-Kubernetes will internally use the
+primary UDN to handle MAC/IP requests from the NSE.
+
+> The default NAD object specified above is already used when the default network is exposed through BGP as
+part of the route advertisement feature. The proposal is to have it available all the time. 
+
+
+With `k8s.ovn.org/primary-udn-ipamclaim` being deprecated in favor of the `IPAMClaimReference` field
+in the `NetworkSelectionElement` we have to define the expected behavior. To avoid conflicting 
+settings when `v1.multus-cni.io/default-network` is set the `k8s.ovn.org/primary-udn-ipamclaim` is 
+going to be ignored, it will be reflected in the opposite scenario for backwards compatibility 
+with a plan to remove it in a future release.
+Deprecation plan for the `k8s.ovn.org/primary-udn-ipamclaim` annotation:
+- release-N - emit a warning event stating that the annotation is deprecated and will be removed in a future release.
+- release-N+1 - fail to configure pods with the annotation set.
+- release-N+2 - remove any code handling the annotation, effectively ignoring it.
+
+Note that `GatewayRequest` is not listed, the default gateway is an attribute of the network is not going to be
+configurable per pod.
+
+### Address allocation
+
+OVN-Kubernetes currently [generates](https://github.com/ovn-kubernetes/ovn-kubernetes/blob/3ef29b9a32b04b7917a0afd6b0e9651d17242ed7/go-controller/pkg/util/net.go#L100-L113)
+the overlay MAC addresses from the IPs:
+
+- IPv4: It takes the four octets of the address (e.g `AA.BB.CC.DD`) and uses them to
+create the MAC address with a constant prefix (e.g. `0A:58:AA:BB:CC:DD`).
+- IPv6: Computes a SHA256 checksum from the IPv6 string and uses the first four bytes for the MAC
+address with the `0A:58` constant prefix(e.g. `0A:58:SHA[0]:SHA[1]:SHA[2]:SHA[3]`).
+
+Although unlikely, we need to implement logic that ensures that the MAC address requested through
```

Comment:
we actually have a downstream bug, that might force us to change our current MAC auto-generation logic. The issue is with the way we generate MACs they are not different enough so that bond hashing always picks the same link in certain hash algorithms. Just raising awareness.